NET-438: Log improvements on WebRTCConnection timeout

bearni95 opened the pull request:
Silences the  connection time-out `logger.warn` on every individual connection and adds a periodic reporting `logger.info` for 'open' and 'connecting' peer states every 5 minutes

bearni95 committed:
fix: silenced connection time out warnings and added periodic reporting of 'open' and 'connecting' peer states

teogeb commented on packages/network/src/connection/WebRtcEndpoint.ts:
```diff
@@ -117,6 +120,29 @@ export class WebRtcEndpoint extends EventEmitter implements IWebRtcEndpoint {
                 return Object.values(this.connections).reduce((total, c) => total + c.getQueueSize(), 0)
             })
+
+        this.statusReportIntervalMs = 5 * 60 * 1000
```
The variable could be a constant as it is not configurable.

teogeb commented on packages/network/src/connection/WebRtcEndpoint.ts:
```diff
@@ -117,6 +120,29 @@ export class WebRtcEndpoint extends EventEmitter implements IWebRtcEndpoint {
+        if (this.statusReportTimeout) {
+            clearTimeout(this.statusReportTimeout)
+        }
+        this.statusReportTimeout = setTimeout(() => {
```
Maybe `setInterval` instead of `setTimeout`, to keep this as simple as possible?

teogeb commented on packages/network/src/connection/WebRtcEndpoint.ts:
```diff
@@ -117,6 +120,29 @@ export class WebRtcEndpoint extends EventEmitter implements IWebRtcEndpoint {
+        }
+        this.statusReportTimeout = setTimeout(() => {
+            let connectedPeerCount = 0
+            const pendingConnectionIds = []
```
Rename to `pendingPeerIds` as these the ids of `peers`, not ids of `connections` (https://github.com/streamr-dev/network-monorepo/blob/14e94a72c4b96a25f306e8e5526061cf8f7c2925/packages/network/src/connection/WebRtcConnection.ts#L125)

bearni95 committed:
fix: corrected comments;

- moved statusReportIntervalMs into a const rather than an object property
- changed the reporting from timeout into interval
- renamed variable from pendingConnectionIds into pendingPeerIds

teogeb commented on packages/network/src/connection/WebRtcEndpoint.ts:
```diff
@@ -49,8 +49,7 @@ export class WebRtcEndpoint extends EventEmitter implements IWebRtcEndpoint {
 
-    private statusReportTimeout?: NodeJS.Timeout
-    private statusReportIntervalMs: number
+    private statusReportInterval?: NodeJS.Timeout
```
The `statusReportInterval` name is maybe misleading, as this variable contains the timer id of `setInterval` call, not the interval length.  Could be e.g.`statusReportTimer`.

teogeb commented on packages/network/src/connection/WebRtcEndpoint.ts:
```diff
@@ -121,28 +120,24 @@ export class WebRtcEndpoint extends EventEmitter implements IWebRtcEndpoint {
-            clearTimeout(this.statusReportTimeout)
-        }
-        this.statusReportTimeout = setTimeout(() => {
+        const statusReportIntervalMs = 5 * 60 * 1000
```
Use the correct naming convention for a constant

bearni95 committed:
fix: renamed const into proper formating; statusReportInterval renamed StatusReportTimer

bearni95 committed:
fix: typo in const name
